pkg/netlink/netlinkstate.go

```diff
@@ -0,0 +1,791 @@
+	} else {
+		nexhthopSt.Resolved = true
+	}
+	return nexhthopSt
+}
+
+// NHAssignID returns the nexthop id
+func NHAssignID(key nexthopKey) int {
+	id := nhIDCache[key]
+	if id == 0 {
+		// Assigne a free id and insert it into the cache
+		id = nhNextID
+		nhIDCache[key] = id
+		nhNextID++
+	}
+	return id
+}
+
+// checkProto checks the proto type
+func checkProto(proto int) bool {
+	var protos = [3]int{unix.RTPROT_BOOT, unix.RTPROT_STATIC, 196}
+	for _, v := range protos {
+		if proto == v {
+			return true
+		}
+	}
+	return false
+}
+
+// checkNeigh checks the nighbor
+func checkNeigh(nk neighKey) bool {
+	for k := range latestNeighbors {
+		if k == nk {
+			return true
+		}
+	}
+	return false
+}
+
+// getProto gets the route protocol
+func getProto(n *RouteStruct) string {
+	for p, i := range rtnProto {
+		if i == int(n.Route0.Protocol) {
+			return p
+		}
+	}
+	return "0"
+}
+
+func printNeigh(neigh *neighStruct) string {
+	var Proto string
+	if neigh == nil {
+		return strNone
+	}
+	if neigh.Protocol == "" {
+		Proto = strNone
+	} else {
+		Proto = neigh.Protocol
+	}
+	str := fmt.Sprintf("Neighbor(vrf=%s dst=%s lladdr=%s dev=%s proto=%s state=%s) ", neigh.VrfName, neigh.Neigh0.IP.String(), neigh.Neigh0.HardwareAddr.String(), NameIndex[neigh.Neigh0.LinkIndex], Proto, getStateStr(neigh.Neigh0.State))
+	return str
+}
+
+// CheckRdup checks the duplication of routes
+func CheckRdup(tmpKey routeKey) bool {
+	var dup = false
+	for j := range latestRoutes {
+		if j == tmpKey {
+			dup = true
+			break
+		}
+	}
+	return dup
+}
+
+// readFDB read the fdb from db
+func readFDB() []*FdbEntryStruct {
+	var fdbs []fdbIPStruct
+	var macs []*FdbEntryStruct
+	var fs *FdbEntryStruct
+
+	CP, err := nlink.ReadFDB(ctx)
+	if err != nil || len(CP) == 3 {
+		return macs
+	}
+
+	var rawMessages []json.RawMessage
+	err = json.Unmarshal([]byte(CP), &rawMessages)
+	if err != nil {
+		log.Printf("JSON unmarshal error: %v", err)
+		return macs
+	}
+	CPs := make([]string, 0, len(rawMessages))
+	for _, rawMsg := range rawMessages {
+		CPs = append(CPs, string(rawMsg))
+	}
+	for i := 0; i < len(CPs); i++ {
+		var fi fdbIPStruct
+		err := json.Unmarshal([]byte(CPs[i]), &fi)
+		if err != nil {
+			log.Printf("netlink: error-%v", err)
+		}
+		fdbs = append(fdbs, fi)
+	}
+	for _, m := range fdbs {
+		fs = ParseFdb(m)
+		if preFilterMac(fs) {
+			macs = append(macs, fs)
+		}
+	}
+	return macs
+}
+
+// ParseFdb parse the fdb
+func ParseFdb(fdbIP fdbIPStruct) *FdbEntryStruct {
+	var fdbentry FdbEntryStruct
+	fdbentry.VlanID = fdbIP.Vlan
+	fdbentry.Mac = fdbIP.Mac
+	fdbentry.Key = fdbKey{fdbIP.Vlan, fdbIP.Mac}
+	fdbentry.State = fdbIP.State
+	fdbentry.Nexthop = &L2NexthopStruct{}
+	lbs, _ := infradb.GetAllLBs()
+	for _, lb := range lbs {
+		if lb.Spec.VlanID == uint32(fdbentry.VlanID) {
+			fdbentry.lb = lb
+			break
+		}
+	}
+	if fdbentry.lb != nil {
+		bp := fdbentry.lb.MacTable[fdbentry.Mac]
+		if bp != "" {
+			fdbentry.bp, _ = infradb.GetBP(bp)
+		}
+	}
+	Dev := fdbIP.Ifname
```

Comment:
why some vars are capitilized, some not? https://google.github.io/styleguide/go/decisions.html

Reply:
Done

Reply:
Done

Reply:
![image](https://github.com/user-attachments/assets/3b7e1eb7-56d1-45be-b522-7e01a438fbd9)

Please check other places

Reply:
Done